### PR TITLE
Add: zh/gptq-integration.md

### PR DESCRIPTION
**Pull request by PanQiWei**
Hi, I've translated gptq-integration.md into Chinese. As the time of this pr, translation on the whole blog is almost done, but I think maybe there are somethings still need to be done before it can be merge?

- [x] Hi @zhongdongy! Could you help to proofread the translations and check if my writting format is correct especially in where Chinese and English appear together?
- [x] I translated "democratize good machine learning" into "让优秀的机器学习大众化" but not sure if it's 100% correct, is there a official Chinese version translation in Hugging Face?
- [ ] Table of contents' indexing may not 100% works.

### CODE REVIEW

File: zh/gptq-integration.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+---
+
+# 使用 AutoGPTQ 和 transformers 让大语言模型更轻量化
+
+<!-- {blog_metadata} -->
+<!-- {authors} -->
+
+大语言模型在理解和生成人类水平的文字方面所展现出的非凡能力，正在许多领域带来应用上的革新。然而，在消费级硬件上训练和部署大语言模型的需求也变得越来越难以满足。
+
+🤗 Hugging Face 的核心使命是 _让优秀的机器学习大众化_ (TODO: 是否有官方正式的中译版本？)，而这正包括了尽可能地让所有人都能够使用上大模型。本着 [与 bitsandbytes 合作](https://huggingface.co/blog/4bit-transformers-bitsandbytes) 一样的精神，我们将 [AutoGPTQ](https://github.com/PanQiWei/AutoGPTQ) 代码库集成到了 Transformers 中，让用户使用 GPTQ 算法 ([Frantar et al. 2023](https://arxiv.org/pdf/2210.17323.pdf)) 在8位、4位、3位，甚至是2位精度下量化和运行模型成为可能。当使用 int4 量化时，精度的下降可以忽略不计，同时在小批量推理上保持着与 `fp16` 基线相当的速度。 需要注意的是，GPTQ 方法与 bitsandbytes 提出的训练后量化方法有所不同：它需要在量化阶段提供一个校准数据集。
```
Comment (xianbaoqian) [inline zh/gptq-integration.md:26]: 普惠化

**Comment by xianbaoqian** [comment]
Thanks for adding the "official" Chinese version. Looks good to me in general with 1 comment.  But I'll also wait for @zhongdongy 's comment if there is any.

File: zh/gptq-integration.md (hunk 1 above)
Comment (PanQiWei) [reply zh/gptq-integration.md]: Thank you for the correction!

File: zh/gptq-integration.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+- 量化模型可被序列化并在 Hugging Face Hub 上分享。
+- GPTQ 方法大大降低运行大语言模型所需的内存，同时保持着与 FP16 相当的推理速度。
+- AutoGPTQ 在更广泛的 transformers 架构上支持 Exllama 算子。
+- 该集成带有基于 RoCm 的 AMD GPU 的本地化支持。
+- 能够 [**使用 PEFT 微调量化后的模型**](#--使用-peft-微调量化后的模型--) 。
+
+你可以在 Hugging Face Hub 上查看你所喜爱的模型是否已经拥有 GPTQ 量化版本。TheBloke，Hugging Face 的顶级贡献者之一，已经使用 AutoGPTQ 量化了大量的模型并分享在 Hugging Face Hub 上。在我们的共同努力下，这些模型仓库都将可以与我们的集成一起开箱即用。
+
+
+以下是一个使用 batch size = 1 的测评结果示例。该测评结果通过在英伟达 A100-SXM4-80GB GPU 上运行得到。我们使用长度为512个词元的提示文本，并精确地生成512个新词元。表格的第一行展示的是未量化的 `fp16` 基线，另外两行则展示使用 AutoGPTQ 不同算子的内存开销和推理性能。
```
Comment (zhongdongy) [inline zh/gptq-integration.md:121]:
```suggestion
以下是一个使用 batch size = 1 的测评结果示例。该测评结果通过在英伟达 A100-SXM4-80GB GPU 上运行得到。我们使用长度为 512 个词元的提示文本，并精确地生成 512 个新词元。表格的第一行展示的是未量化的 `fp16` 基线，另外两行则展示使用 AutoGPTQ 不同算子的内存开销和推理性能。
```

File: zh/gptq-integration.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+from transformers import AutoModelForCausalLM, AutoTokenizer, GPTQConfig
+
+model_id = "facebook/opt-125m"
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+quantization_config = GPTQConfig(bits=4, dataset = "c4", tokenizer=tokenizer)
+
+model = AutoModelForCausalLM.from_pretrained(model_id, device_map="auto", quantization_config=quantization_config)
+```
+
+量化一个模型可能花费较长的时间。对于一个 175B 参数量的模型，如果使用一个大型校准数据集（如 “c4”），至少需要4个 GPU 时。正如上面提到的那样，许多 GPTQ 模型已经可以在 Hugging Face Hub 上被取得，这让你在大多数情况下无需自行量化模型。当然，你仍可以使用你所专注的特定领域的数据集来量化模型。
```
Comment (zhongdongy) [inline zh/gptq-integration.md:148]:
```suggestion
量化一个模型可能花费较长的时间。对于一个 175B 参数量的模型，如果使用一个大型校准数据集（如 “c4”），至少需要 4 个 GPU 时。正如上面提到的那样，许多 GPTQ 模型已经可以在 Hugging Face Hub 上被取得，这让你在大多数情况下无需自行量化模型。当然，你仍可以使用你所专注的特定领域的数据集来量化模型。
```

File: zh/gptq-integration.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+例如，这一集成允许在单个 A100-80GB GPU上服务 70B 模型！而这在使用 fp16 的模型权重时是不可能的，因为它超出了最大可用的 GPU 内存。
+
+你可以在 TGI 的 [说明文档](https://huggingface.co/docs/text-generation-inference/main/en/basic_tutorials/preparing_model#quantization) 中找到更多有关 GPTQ 的用法。
+
+需要注意的时，TGI 中集成的算子不能很好地扩展到较大的批处理大小。因此，这一方式虽然节省了内存，但在较大的批处理大小上发生速度的下降是符合预期的。
+
+## **使用 PEFT 微调量化后的模型**
+
+在常规的方法下，你无法进一步微调量化后的模型。然而，通过使用 PEFT 代码库，你可以在量化后的模型之上训练适应性网络！为实现这一目标，我们冻结了量化过的基座模型的所有网络层，并额外添加可训练的适应性网络。这里是一些关于如何使用 PEFT 训练 GPTQ 模型的例子：[colab 笔记本](https://colab.research.google.com/drive/1VoYNfYDKcKRQRor98Zbf2-9VQTtGJ24k?usp=sharing) 和 [微调脚本](https://gist.github.com/SunMarc/dcdb499ac16d355a8f265aa497645996) 。
```
Comment (zhongdongy) [inline zh/gptq-integration.md:162]:
```suggestion
在常规的方法下，你无法进一步微调量化后的模型。然而，通过使用 PEFT 代码库，你可以在量化后的模型之上训练适应性网络！为实现这一目标，我们冻结了量化过的基座模型的所有网络层，并额外添加可训练的适应性网络。这里是一些关于如何使用 PEFT 训练 GPTQ 模型的例子：[Colab 笔记本](https://colab.research.google.com/drive/1VoYNfYDKcKRQRor98Zbf2-9VQTtGJ24k?usp=sharing) 和 [微调脚本](https://gist.github.com/SunMarc/dcdb499ac16d355a8f265aa497645996) 。
```

File: zh/gptq-integration.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+## 结论和结语
+
+本文中，我们介绍了 Transformers 对 [AutoGPTQ 代码库](https://github.com/PanQiWei/AutoGPTQ) 的集成，使得社区中的任何人都可以更方便地利用 GPTQ 方法量化大语言模型，助力令人激动的大语言模型工具和应用的构建。
+
+这一集成支持英伟达 GPU 和基于 RoCm 的 AMD GPU，这是向支持更广泛 GPU 架构的量化模型的普惠化迈出的一大步。
+
+与 AutoGPTQ 团队的合作非常富有成效，我们非常感谢他们的支持和他们在该代码库上的工作。
+
+我们希望本次集成将使每个人都更容易地在他们的应用程序中使用大语言模型，我们迫不及待地看到大家即将使用它所创造出的一切！
```
Comment (zhongdongy) [inline zh/gptq-integration.md:188]:
```suggestion
我们希望本次集成将使每个人都更容易地在他们的应用程序中使用大语言模型，我们迫不及待地想要看到大家即将使用它所创造出的一切！
```

File: zh/gptq-integration.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+---
+
+# 使用 AutoGPTQ 和 transformers 让大语言模型更轻量化
+
+<!-- {blog_metadata} -->
+<!-- {authors} -->
+
+大语言模型在理解和生成人类水平的文字方面所展现出的非凡能力，正在许多领域带来应用上的革新。然而，在消费级硬件上训练和部署大语言模型的需求也变得越来越难以满足。
+
+🤗 Hugging Face 的核心使命是 _让优秀的机器学习普惠化_ ，而这正包括了尽可能地让所有人都能够使用上大模型。本着 [与 bitsandbytes 合作](https://huggingface.co/blog/4bit-transformers-bitsandbytes) 一样的精神，我们将 [AutoGPTQ](https://github.com/PanQiWei/AutoGPTQ) 代码库集成到了 Transformers 中，让用户使用 GPTQ 算法 ([Frantar et al. 2023](https://arxiv.org/pdf/2210.17323.pdf)) 在8位、4位、3位，甚至是2位精度下量化和运行模型成为可能。当使用 int4 量化时，精度的下降可以忽略不计，同时在小批量推理上保持着与 `fp16` 基线相当的速度。 需要注意的是，GPTQ 方法与 bitsandbytes 提出的训练后量化方法有所不同：它需要在量化阶段提供一个校准数据集。
```
Comment (zhongdongy) [inline zh/gptq-integration.md:26]:
```suggestion
🤗 Hugging Face 的核心使命是 _让优秀的机器学习普惠化_ ，而这正包括了尽可能地让所有人都能够使用上大模型。本着 [与 bitsandbytes 合作](https://huggingface.co/blog/4bit-transformers-bitsandbytes) 一样的精神，我们将 [AutoGPTQ](https://github.com/PanQiWei/AutoGPTQ) 代码库集成到了 Transformers 中，让用户使用 GPTQ 算法 ([Frantar et al. 2023](https://arxiv.org/pdf/2210.17323.pdf)) 在 8 位、4 位、3 位，甚至是 2 位精度下量化和运行模型成为可能。当使用 int4 量化时，精度的下降可以忽略不计，同时在小批量推理上保持着与 `fp16` 基线相当的速度。 需要注意的是，GPTQ 方法与 bitsandbytes 提出的训练后量化方法有所不同：它需要在量化阶段提供一个校准数据集。
```

File: zh/gptq-integration.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+具体而言，GPTQ 采用 int4/fp16 (W4A16) 的混合量化方案，其中模型权重被量化为 int4 数值类型，而激活值则保留在 float16。在推理阶段，模型权重被动态地反量化回 float16 并在该数值类型下进行实际的运算。
+
+该方案有以下两方面的优点：
+
+- int4 量化能够节省接近4倍的内存，这是因为反量化操作发生在算子的计算单元附近，而不是在 GPU 的全局内存中。
+- 由于用于权重的位宽较低，因此可以节省数据通信的时间，从而潜在地提升了推理速度。
+
+GPTQ 论文解决了分层压缩的问题：
+
+给定一个拥有权重矩阵 \\(W_{l}\\) 和输入 \\(X_{l}\\) 的网络层 \\(l\\)，我们期望获得一个量化版本的权重矩阵 \\(\hat{W}_{l}\\) 以最小化均方误差（MSE）：
```
Comment (zhongdongy) [inline zh/gptq-integration.md:74]:
```suggestion
给定一个拥有权重矩阵 \\(W_{l}\\) 和输入 \\(X_{l}\\) 的网络层 \\(l\\)，我们期望获得一个量化版本的权重矩阵 \\(\hat{W}_{l}\\) 以最小化均方误差 (MSE)：
```

File: zh/gptq-integration.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+
+为解决这一分层压缩问题，论文作者采用了最优脑量化 (Optimal Brain Quantization, OBQ) 框架 ([Frantar et al 2022](https://arxiv.org/abs/2208.11580)) 。OBQ 方法的出发点在于其观察到：以上等式可以改写成权重矩阵 \\(W_{l}\\) 每一行的平方误差之和
+
+\\( \sum_{i=0}^{d_{row}} \|W_{l[i,:]}X-\hat{W}_{l[i,:]}X\|^{2}_{2} \\)
+
+这意味着我们可以独立地对每一行执行量化。即所谓的 per-channel quantization。对每一行 \\(W_{l[i,:]}\\)，OBQ 在每一时刻只量化一个权重，同时更新所有未被量化的权重，以补偿量化单个权重所带来的误差。所选权重的更新采用一个闭环公式，并利用了海森矩阵 (Hessian Matrices)。
+
+GPTQ 论文通过引入一系列优化措施来改进上述量化框架，在降低量化算法复杂度的同时保留了模型的精度。
+
+相较于 OBQ，GPTQ 的量化步骤本身也更快：OBQ 需要花费2个 GPU 时来完成 BERT 模型 (336M) 的量化，而使用 GPTQ，量化一个 Bloom 模型 (176B) 则只需不到4个 GPU 时。
```
Comment (zhongdongy) [inline zh/gptq-integration.md:88]:
```suggestion
相较于 OBQ，GPTQ 的量化步骤本身也更快：OBQ 需要花费 2 个 GPU 时来完成 BERT 模型 (336M) 的量化，而使用 GPTQ，量化一个 Bloom 模型 (176B) 则只需不到 4 个 GPU 时。
```

File: zh/gptq-integration.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+- user: PanEa
+  guest: true
+- user: qwopqwop
+  guest: true
+- user: ybelkada
+- user: TheBloke
+  guest: true
+translators:
+- user: PanEa
+  guest: true
```
Comment (zhongdongy) [inline zh/gptq-integration.md:16]:
```suggestion
  guest: true
- user: zhongdongy
  proofreader: true
```

**Review by zhongdongy — APPROVED** [review]
@PanQiWei Thanks for your perfect translation and @xianbaoqian thanks for your suggestion. I adjusted several formattings and added proofreader tag in meta data. I think it's good enough to merge! 🌹

**Pull request merged**